include/posix/sys/types.h

```diff
@@ -76,6 +76,39 @@ typedef struct pthread_rwlock_obj {
 	k_tid_t wr_owner;
 } pthread_rwlock_t;
 
+typedef u32_t pthread_once_t;
+
+/* pthread_key */
+typedef void *pthread_key_t;
+
+typedef struct pthread_key_obj {
+	/* Optional destructor that is passed to pthread_key_create() */
+	void (*destructor)(void *);
+
+	/* List of KeyData objects that contain thread
+	 * specific data for the key
+	 */
+	sys_dlist_t KeyDataL;
+} pthread_key_obj;
+
+typedef struct ThreadData {
+	sys_dnode_t node;
+
+	/* Key and thread specific data passed to pthread_setspecific() */
+	pthread_key_obj *key;
+	void *specData;
+
+	/* Back pointer for easy removal of thread data from KeyDataL,
+	 * the key's thread data list, when the thread exits
+	 */
+	struct KeyData *keyData;
```

Comment:
It might be OK to not have this pointer and traverse `KeyDataL` when thread exits.  Threads exiting are not that common, and this saves a pointer from this struct.

Reply:
Okay, I have removed it here, and also in `pthread_setspecific`, where it is initialized.